src/resdk/tables/variant.py

```diff
@@ -99,70 +99,118 @@ def __init__(
 
         self.discard_fakes = discard_fakes
 
-        self.mutations = mutations
-        if mutations is not None:
-            if not isinstance(mutations, list):
-                raise ValueError("Mutations must be a list of strings")
+        self._geneset = None
+        if geneset is None:
+            self._check_heterogeneous_mutations()
         else:
-            # Check for heterogeneous mutations only if mutations is not specified
-            self.check_heterogeneous_mutations()
+            self.geneset = geneset
 
-    def check_heterogeneous_mutations(self):
-        """Ensure variants are selected for the same mutations in all samples."""
-        mutations = {str(d.input["mutations"]) for d in self._data}
+    @property
+    def geneset(self):
+        """Get geneset."""
+        return self._geneset
+
+    @geneset.setter
+    def geneset(self, value):
+        """Set geneset."""
+        # Geneset can be set only once, prevent modifications
+        if self._geneset is not None:
+            raise ValueError("It is not allowed to change geneset value.")
+
+        if value is None:
+            return
+
+        # If id / slug of a geneset is given, get it from the Resolwe server
+        if isinstance(value, (int, str)):
+            gs = self.resolwe.data.get(value)
+            value = gs.genes
+        elif isinstance(value, Geneset):
+            value = value.genes
+
+        if isinstance(value, (list, set, tuple, pd.Series)):
+            self._geneset = set(value)
+        else:
+            raise ValueError(f'Unsupported type of "geneset" input: {value}.')
+
+    @property
+    @lru_cache()
+    def variants(self) -> pd.DataFrame:
+        """Return variants table as a pandas DataFrame object."""
+        return self._load_fetch(self.VARIANTS)
+
+    @property
+    @lru_cache()
+    def depth(self) -> pd.DataFrame:
+        """Return depth table as a pandas DataFrame object."""
+        return self._load_fetch(self.DEPTH)
+
+    @property
+    @lru_cache()
+    def depth_a(self) -> pd.DataFrame:
+        """Return depth table for adenine as a pandas DataFrame object."""
+        return self._load_fetch(self.DEPTH_A)
+
+    @property
+    @lru_cache()
+    def depth_c(self) -> pd.DataFrame:
+        """Return depth table for cytosine as a pandas DataFrame object."""
+        return self._load_fetch(self.DEPTH_C)
+
+    @property
+    @lru_cache()
+    def depth_g(self) -> pd.DataFrame:
+        """Return depth table for guanine as a pandas DataFrame object."""
+        return self._load_fetch(self.DEPTH_G)
+
+    @property
+    @lru_cache()
+    def depth_t(self) -> pd.DataFrame:
+        """Return depth table for thymine as a pandas DataFrame object."""
+        return self._load_fetch(self.DEPTH_T)
+
```

Comment:
Add additional function to compute dataframe based on column `FILTER` from input dataframes. The FILTER column tells us if the variant passed variant filtration, so something like `tables.qc` could be used to return the dataframe. Values in FILTER column are strings.